libs/community/langchain_community/utilities/astradb.py

```diff
@@ -46,39 +48,115 @@ def __init__(
 
         if token and api_endpoint:
             astra_db = AstraDB(
-                token=self.token,
-                api_endpoint=self.api_endpoint,
+                token=token,
+                api_endpoint=api_endpoint,
+                namespace=self.namespace,
+            )
+            async_astra_db = AsyncAstraDB(
+                token=token,
+                api_endpoint=api_endpoint,
                 namespace=self.namespace,
             )
-            if supports_async:
-                self.async_astra_db = db.AsyncAstraDB(
-                    token=self.token,
-                    api_endpoint=self.api_endpoint,
-                    namespace=self.namespace,
-                )
 
         if astra_db:
             self.astra_db = astra_db
+            if async_astra_db:
+                self.async_astra_db = async_astra_db
+            else:
+                self.async_astra_db = AsyncAstraDB(
+                    token=self.astra_db.token,
+                    api_endpoint=self.astra_db.base_url,
+                    api_path=self.astra_db.api_path,
+                    api_version=self.astra_db.api_version,
+                    namespace=self.astra_db.namespace,
+                )
+        elif async_astra_db:
+            self.async_astra_db = async_astra_db
+            self.astra_db = AstraDB(
+                token=self.async_astra_db.token,
+                api_endpoint=self.async_astra_db.base_url,
+                api_path=self.async_astra_db.api_path,
+                api_version=self.async_astra_db.api_version,
+                namespace=self.async_astra_db.namespace,
+            )
+        else:
+            raise ValueError(
+                "Must provide 'astra_db_client' or 'async_astra_db_client' or "
+                "'token' and 'api_endpoint'"
+            )
+
+
+class AstraDBCollectionEnvironment(AstraDBEnvironment):
```

Comment:
* Would you mind adding doc-strings explaining what this is?
* Could we prefix this with a `_` to mark is as an internal object? I don't think that we expect most LangChain users to know about this object?

Reply:
done

Reply:
ah it was public because it's used in other files (obvious, but I missed that lol). I don't think we have a great convention right now with respect to public/private. I still prefer erring on the side of private and prioritize what users see as public